frontend/src/scenes/saved-insights/savedInsightsLogic.ts

```diff
@@ -236,7 +236,17 @@ export const savedInsightsLogic = kea<savedInsightsLogicType<InsightsResult, Sav
         }
     },
     urlToAction: ({ actions, values }) => ({
-        '/saved_insights': (_, searchParams) => {
+        [urls.savedInsights()]: (_, searchParams, hashParams) => {
+            if (hashParams.fromItem) {
+                // `fromItem` for legacy /insights url redirect support
```

Comment:
This gives us backwards compatibility for `fromItem` URLs, but those without `fromItem` don't work anymore. I think it may be retaining support for this to avoid frustrating users who have old links. Though that would probably require creating a new insight record every time such a link is accessed (`createAndRedirectToNewInsight`-style), which is not ideal. Your call, let's just make the decision explicit.

Reply:
Fixed!